core/docker_service_linux.go

```diff
@@ -0,0 +1,21 @@
+/*
+Copyright 2021 Mirantis
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import "github.com/docker/docker/pkg/parsers/kernel"
+
+var kernelSupportsRRO = kernel.CheckKernelVersion(5, 12, 0)
```

Comment:
1. Why is the CRI checking if a kernel supports a feature? Shouldn't it be the runtime's responsibility to report the `rro` feature iff the kernel supports it?
2. We can't blindly assume that kernels with a version number < 5.12.0 do not support RRO mounts. Distro kernels might have the feature backported.
3. We can't assume that the kernel hosting the CRI is the same as the kernel hosting the container. The runtime might use VM-based isolation, such as Kata. Or someone might be crazy enough to configure cri-dockerd to connect to a remote dockerd on another host.

Reply:
> 1. Why is the CRI checking if a kernel supports a feature? 

Because it is CRI runtime's responsibility.
Same as containerd and CRI-O.


> Shouldn't it be the runtime's responsibility to report the `rro` feature iff the kernel supports it?

No: https://github.com/opencontainers/runtime-spec/blob/v1.2.0/features.md?plain=1#L6-L7
```
The Features structure is irrelevant to the actual availability of the features in the host operating system.
Hence, the content of the Features structure SHOULD be determined on the compilation time of the runtime, not on the execution time.
```


> 2. We can't blindly assume that kernels with a version number < 5.12.0 do not support RRO mounts. Distro kernels might have the feature backported.

There is no well-known distro that has backported RRO.
Even if there is, we do not need to actively care about such a "frankenized" kernel unless it has an exceptionally huge number of users.

> 3. We can't assume that the kernel hosting the CRI is the same as the kernel hosting the container. The runtime might use VM-based isolation, such as Kata. 

Kata can't support RRO, as it can't distinguish RRO from RO.
In Kata, just a plain RO works as RRO.

> Or someone might be crazy enough to configure cri-dockerd to connect to a remote dockerd on another host.

I don't think this configuration is supported.
No mount will work anyway.